packages/models/src/catalogItem.ts

```diff
@@ -54,12 +54,12 @@ const descriptor = z.object({
   dailyCallsPerConsumer: z.number().int(),
   dailyCallsTotal: z.number().int(),
   agreementApprovalPolicy: z.enum(["MANUAL", "AUTOMATIC"]).optional(),
-  createdAt: z.string().datetime(),
+  createdAt: z.date(),
   serverUrls: z.array(z.string()),
-  publishedAt: z.string().datetime().optional(),
-  suspendedAt: z.string().datetime().optional(),
-  deprecatedAt: z.string().datetime().optional(),
-  archivedAt: z.string().datetime().optional(),
+  publishedAt: z.date().optional(),
+  suspendedAt: z.date().optional(),
+  deprecatedAt: z.date().optional(),
+  archivedAt: z.date().optional(),
```

Comment:
If we use this decoder to decode the readmodel from the mongodb we could have problems. Maybe `z.coerce.date()` is better?

Reply:
👍🏻  Ok to use `z.coerce` at the moment, but we need to unify all data formats and conversions, so as discussed offline we should convert it properly in future tasks.